spec/unit/provider_spec.rb

```diff
@@ -436,6 +436,70 @@ def existing_command
       expect(child.specificity).to be > parent.specificity
     end
 
+    ### TODO: A bunch of tests with mulitple defaultfor and notdefaultfor testing specificity
+    ## My intuition: if you don't like jelly beans, and I don't like green jelly beans, I am more likely to eat skittles?
+
+    #Competing default only different in notdefaultfor
+    it "should choose a more specific non-default over a less specific" do
+      skip("Not ready for prime time")
+      Facter.expects(:value).with(:osfamily).at_least_once.returns "redhat"
+      Facter.expects(:value).with(:operatingsystem).at_least_once.returns "centos"
+      Facter.expects(:value).with(:operatingsystemrelease).at_least_once.returns "28"
+
+      one = type.provide(:one) do
+        defaultfor :osfamily => "redhat"
+        notdefaultfor :operatingsystem => "fedora"
+      end
+
+      two = type.provide(:two) do
+        defaultfor :operatingsystem => "redhat"
+        notdefaultfor :operatingsystem => "fedora", :operatingsystemrelease => "27"
+      end
+
+      expect(type.defaultprovider).to eq(two)
+    end
+
+    #Competing not defaults only different in defaultfor
+    it "should choose a more specific non-default over a less specific" do
+      skip("Not ready for prime time")
+      Facter.expects(:value).with(:osfamily).at_least_once.returns "redhat"
+      Facter.expects(:value).with(:operatingsystem).at_least_once.returns "fedora"
+      Facter.expects(:value).with(:operatingsystemrelease).at_least_once.returns "27"
+
+      one = type.provide(:one) do
+        defaultfor :osfamily => "redhat"
+        notdefaultfor :operatingsystem => "fedora"
+      end
+
+      two = type.provide(:two) do
+        defaultfor :operatingsystem => "redhat"
+        notdefaultfor :operatingsystem => "fedora", :operatingsystemrelease => "27"
+      end
+
+      expect(type.defaultprovider).to eq(two)
+    end
+
+    #Competing not defaults only different in defaults
+    it "should choose a more specific non-default over a less specific" do
```

Comment:
Shouldn't this be using the class inheritance rule, since none of the `defaultfor`, or `notdefaultfor` match what Facter has been stubbed out to return?